### PR TITLE
feat(azure): add support for Azure Cognitive Search

### PR DESCRIPTION
**Pull request by ekifel**
Issue #790

### CODE REVIEW

**Comment by tim775** [comment]
@alikhajeh1 LGTM

**Comment by ekifel** [comment]
I checked cost [estimate](https://azure.com/e/77d0df55f62d4715b4229cc9990a7292), `Search usage` cost components are correct, but `Image extraction` doesn't have a support in Azure pricing calculator.

File: internal/providers/terraform/azure/search_service.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	}
+	units := decimalPtr(partitionCount.Mul(replicaCount))
+
+	var skuName string
+	skuElems := strings.Split(sku, " ")
+	for _, v := range skuElems {
+		skuName += strings.Title(v) + " "
+	}
+	costComponents = append(costComponents, &schema.CostComponent{
+		Name:           fmt.Sprintf("Search usage (%s, %s unit)", skuName[:len(skuName)-1], units.String()),
```
Comment (alikhajeh1) [inline internal/providers/terraform/azure/search_service.go:59]: can we make it `1 unit` and `2 units` to handle singular/plural?

**Comment by ekifel** [comment]
@alikhajeh1 updated!

**Pull request merged**